Makefile

```diff
@@ -57,7 +57,7 @@ fix: ## Fixup files in the repo.
 
 .PHONY: setup-lint
 setup-lint: ## Setup the lint
-	$(SCRIPTS_DIR)/fetch golangci-lint 1.50.0
+	$(SCRIPTS_DIR)/fetch golangci-lint 1.51.2
```

Comment:
Since we are on this, there were a bunch of deprecated linters which we seem to still use.

```
WARN [runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'.
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.
WARN [runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.
WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
```
Could we also replace the available alternative ones in https://github.com/operator-framework/operator-sdk/blob/master/.golangci.yml.

Reply:
Done in commit https://github.com/operator-framework/operator-sdk/pull/6351/commits/ebf16c8e09be81c419c1d2fc15529c03a43c5f56